ludwig/commit.py

```diff
@@ -0,0 +1,112 @@
+import argparse
+import logging
+import sys
+from typing import Optional
+
+from ludwig.utils.commit_utils import HuggingFaceHub
+from ludwig.utils.print_utils import get_logging_level_registry
+
+logger = logging.getLogger(__name__)
+
+
+def commit_cli(
+    repo_id: str,
+    model_path: str,
+    repo_type: str = "model",
+    private: bool = False,
+    commit_message: str = "Upload trained [Ludwig](https://ludwig.ai/latest/) model weights",
+    commit_description: Optional[str] = None,
+    **kwargs,
+) -> None:
+    """Create an empty repo on the HuggingFace Hub and upload trained model artifacts to that repo.
+
+    Args:
+        repo_id (`str`):
+            A namespace (user or an organization) and a repo name separated
+            by a `/`.
+        model_path (`str`):
+            The path of the saved model. This is the top level directory where
+            the models weights as well as other associated training artifacts
+            are saved.
+        private (`bool`, *optional*, defaults to `False`):
+            Whether the model repo should be private.
+        repo_type (`str`, *optional*):
+            Set to `"dataset"` or `"space"` if uploading to a dataset or
+            space, `None` or `"model"` if uploading to a model. Default is
+            `None`.
+        commit_message (`str`, *optional*):
+            The summary / title / first line of the generated commit. Defaults to:
+            `f"Upload {path_in_repo} with huggingface_hub"`
+        commit_description (`str` *optional*):
+            The description of the generated commit
+    """
+
```

Comment:
I am wondering if we should allow to publish to our internal ludwig model-hub as well

Reply:
We can!